docs/source/dev-guide/deep-dive-context.md

```diff
@@ -0,0 +1,133 @@
+performed. If everything goes well, you obtain your values just fine. If not, the validation errors
+are raised.
+
+Take into account that the result is cached for faster subsequent access. This means that even
+if you change the value of the environment variables responsible for a given setting, this won't be
+reflected in the `context` object until you refresh it with `conda.base.context.reset_context()`.
+
+```{admonition} Do not modify the Context object!
+
+`ParameterLoader` does not implement the `__set__` method of the `property` protocol, so you
+can freely override an attribute defined in a `Configuration` subclass. You might think that
+this will redefine the value after passing through the validation machinery, but that's not true.
+You will simply overwrite it entirely with the raw value and that's probably not what you want.
+
+Instead, consider the `context` object immutable. If you need to change a setting at runtime, it is
+probably _A Bad Idea_. The only situation where this is acceptable is during testing.
+```
+
+## Setting values in the different origins
+
+There's some magic behind the possible origins for the settings values. How these are tied to the
+final `Configuration` object might not be obvious at first. This is different for each
+`RawParameter` subclass:
+
+* `DefaultValueRawParameter`: Users will never see this one. It only wraps the default value passed
+  to the `ParameterLoader` class. Safe to ignore.
+* `YamlRawParameter`: This one takes a YAML file and parses it as a dictionary. The keys in this
+  file must match the attribute names in the `Configuration` class exactly (or one of their
+  aliases). Matching happens automatically once this is properly set up. How the values are parsed
+  depends on the YAML Loader, set internally by `conda`.
+* `EnvRawParameter`: Values coming from certain environment variables can make it to the
+  `Configuration` instance, provided they are formatted as `<APP_NAME>_<PARAMETER_NAME>`, all
+  uppercase. The app name is defined by the `Configuration` subclass. The parameter name is
+  defined by the attribute name in the class, transformed to upper case. For example,
+  `context.ignore_pinned` can be set with `CONDA_IGNORE_PINNED`. The value of the variable is parsed
+  in different ways depending on the type:
+    * `PrimitiveParameter` is the easy one. The environment variable string is parsed as the
+      expected type. Booleans are a bit different since several strings are recognized as such, and
+      in a case-insensitive way:
+        * `True` can be set with `true`, `yes`, `on` and `y`.
+        * `False` can be set with `false`, `off`, `n`, `no`, `non`, `none` and `""` (empty string).
+    * `SequenceParameter` can specify their own delimiter (e.g. `,`), so the environment variable
+      string is processed into a list.
+    * `MapParameter` and `ObjectParameter` do not seem to support beign set with environment
```

Comment:
Maybe change this sentence to: 
"`MapParameter` and `ObjectParameter` do not support being set with environment variables" 
(corrected misspelling of `being` and removed ambiguity of if it's possible to do this or not)